core/src/main/java/io/grpc/internal/DelayedClientTransport.java

```diff
@@ -307,8 +307,9 @@ boolean isInBackoffPeriod() {
    */
   void startBackoff(final Status status) {
     synchronized (lock) {
-      Preconditions.checkState(!inBackoffPeriod);
-      inBackoffPeriod = true;
+      Preconditions.checkState(backoffStatus == null);
+      backoffStatus = Status.UNAVAILABLE.withDescription("Channel in TRANSIENT_FAILURE state")
+          .withCause(status.asRuntimeException());
```

Comment:
I went ahead and added a new test to `DelayedClientTransportTest` to check behavior on new streams (and verify the status). I also added to TransportSetTest to have it check one of the statuses to make sure that it was propagated from transportShutdown.